crates/astria-sequencer/src/state_ext.rs

```diff
@@ -46,8 +47,19 @@ pub(crate) trait StateReadExt: StateRead {
 
     #[instrument(skip(self))]
     async fn get_revision_number(&self) -> Result<u64> {
-        // this is used for chain upgrades, which we do not currently have.
-        Ok(0)
+        let Some(bytes) = self
+            .get_raw(REVISION_NUMBER_KEY)
+            .await
+            .context("failed to read raw revision number from state")?
+        else {
+            bail!("revision number not found in state");
+        };
+
+        let Ok(bytes): Result<[u8; 8], _> = bytes.try_into() else {
```

Comment:
You can report the actual number of bytes (`map_err` with `anyhow` instead of `context` to get access to to the `Vec` returned by `try_from`: https://doc.rust-lang.org/stable/std/primitive.array.html#method.try_from-5)

```rust
let bytes = TryInto::<[u8; 8]>::try_into(bytes)
    .map_err(|b| anyhow!("expected 8 revision number bytes but got{}", b.len()))?;
```